### PR TITLE
querydsl.variableNameFunctionClass swallows ClassNotFoundExceptions

### PR DESCRIPTION
**Pull request by Shredder121**
I was either going to introduce a new try-catch around the `Class.forName` or adding a catch clause to the existing try-catch.
Which one do you prefer?

### CODE REVIEW

File: querydsl-apt/src/main/java/com/querydsl/apt/DefaultConfiguration.java [hunk 1]
```diff
@@ -195,6 +196,8 @@ public DefaultConfiguration(
                 variableNameFunction = variableNameFunctionClass.newInstance();
             } catch (RuntimeException e) {
                 throw e;
+            } catch (ClassNotFoundException e) {
+                throw Throwables.propagate(e);
```
Comment (natros) [inline querydsl-apt/src/main/java/com/querydsl/apt/DefaultConfiguration.java]: Guava intends to deprecate this method, and recommends throwing a RuntimeException directly
https://github.com/google/guava/blob/2bcf189a62275bb23fd92edf4dc72e9ad849a89e/guava/src/com/google/common/base/Throwables.java#L201-L203

Comment (Shredder121) [inline querydsl-apt/src/main/java/com/querydsl/apt/DefaultConfiguration.java]: Thanks for the heads up!
It's still a WIP, but I appreciate the comment.

**Comment by timowest** [comment]
The proposed solution is fine

**Comment by Shredder121** [comment]
Shall we maybe for Querydsl 5 (or now) add a NamingFunction class, so people can just implement that?
So you can be sure you get the generics right?
Just an idea.

**Comment by timowest** [comment]
NamingFunction sounds good, it could extend `Function<EntityType, String>`, that way we could introduce it already now.

File: querydsl-apt/src/main/java/com/querydsl/apt/DefaultConfiguration.java [hunk 2]
```diff
@@ -195,6 +195,8 @@ public DefaultConfiguration(
                 variableNameFunction = variableNameFunctionClass.newInstance();
             } catch (RuntimeException e) {
                 throw e;
+            } catch (ClassNotFoundException e) {
+                throw new RuntimeException(e);
             } catch (Exception e) {
                variableNameFunction = DefaultVariableNameFunction.INSTANCE;
```
Comment (Shredder121) [inline querydsl-apt/src/main/java/com/querydsl/apt/DefaultConfiguration.java]: Should we also log this?
Or write it the syserr?

Comment (Shredder121) [inline querydsl-apt/src/main/java/com/querydsl/apt/DefaultConfiguration.java]: I mean the `Exception`.

File: querydsl-apt/src/main/java/com/querydsl/apt/NamingFunction.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+import com.google.common.base.Function;
+import com.querydsl.codegen.EntityType;
+
+/**
+ * The {@link Function} signature to convert {@link EntityType} to {@link String}.
+ *
+ * @author Shredder121
+ */
+public interface NamingFunction extends Function<EntityType, String> {
```
Comment (timowest) [inline querydsl-apt/src/main/java/com/querydsl/apt/NamingFunction.java]: should we change the existing naming functions to implement this interface?

Comment (Shredder121) [inline querydsl-apt/src/main/java/com/querydsl/apt/NamingFunction.java]: That's up to you.
We'll probably have to add a new `catch` for `CCE` and a friendly notice that we moved to a better interface :wink:

**Comment by timowest** [comment]
Looks good

**Comment by Shredder121** [comment]
Actually, I only now see what you mean.
For that we would need to move the interface to Querydsl Codegen.
We can still do that, now that it's not yet merged.

**Comment by timowest** [comment]
Great, this was what I had in mind

**Pull request merged**